### PR TITLE
osx-trash 0.7.1 (new formula)

### PR DESCRIPTION
**Pull request by danielbayley**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/osx-trash.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  url "https://github.com/morgant/tools-osx.git",
+      tag:      "trash-0.7.1",
+      revision: "0c8de9c0f92cc035c63ada1ec3943f5263330807"
+  license "MIT"
+  head "https://github.com/morgant/tools-osx.git", branch: "master"
+
+  livecheck do
+    url "https://github.com/morgant/tools-osx/tags"
+    strategy :page_match
+    regex(/trash-(\d+(?:\.\d+)+)/i)
```
Comment (nandahkrishna) [inline Formula/osx-trash.rb:13]:
```suggestion
    url :stable
    regex(/^trash[._-]v?(\d+(?:\.\d+)+)$/i)
```

**Review by nandahkrishna — COMMENTED** [review]
This should conflict with `trash`, `trash-cli` and `macos-trash`.

**Comment by nandahkrishna** [comment]
It seems like there's a bunch of programs/tools in the repository, any particular reason you want to provide just this one?

File: Formula/osx-trash.rb (hunk 1 above)
Comment (danielbayley) [reply Formula/osx-trash.rb:13]: Ha yeah, was already on it 😅

**Comment by danielbayley** [comment]
> It seems like there's a bunch of programs/tools in the repository, any particular reason you want to provide just this one?

I don’t want them all personally and think they should probably be installable separately. Then maybe a `tools-osx` formula that does install them all…

**Comment by danielbayley** [comment]
> This should conflict with `trash`, `trash-cli` and `macos-trash`.

Ha yeah, was already on it 😅

**Comment by carlocab** [comment]
You should add the conflicts in a separate PR after this one is finished so we don't have to needlessly rebuild multiple formulae.

**Comment by danielbayley** [comment]
> You should add the conflicts in a separate PR after this one is finished so we don't have to needlessly rebuild multiple formulae.

Ah ok, will do…

**Comment by danielbayley** [comment]
Looks like the [`test`](https://github.com/Homebrew/homebrew-core/pull/123002/files#diff-1c40406ea4bd244470d603031acfd62dc16f7d99fa1ec3651871a17bfd34bb75R21) [requires](https://github.com/Homebrew/homebrew-core/actions/runs/4151971517/jobs/7182644296#step:6:81) System Preferences > Security & Privacy > Privacy > Full Disk Access. Not sure what to do about that…?

img

**Comment by ZhongRuoyu** [comment]
> Looks like the [`test`](https://github.com/Homebrew/homebrew-core/pull/123002/files#diff-1c40406ea4bd244470d603031acfd62dc16f7d99fa1ec3651871a17bfd34bb75R21) [requires](https://github.com/Homebrew/homebrew-core/actions/runs/4151971517/jobs/7182644296#step:6:81) System Preferences > Security & Privacy > Privacy > Full Disk Access. Not sure what to do about that…?

Unfortunately, this limit cannot be bypassed.

**Comment by danielbayley** [comment]
> Unfortunately, this limit cannot be bypassed.

So there is _no_ other formula that requires any kind of accessibility? Would this then have to be a Cask instead?

**Comment by ZhongRuoyu** [comment]
> So there is _no_ other formula that requires any kind of accessibility?

It's not that there's no other formula requiring such permissions. Rather, to my knowledge, no formula here won't run without that. (All formula should at least come with a passing test, so if any of them failed the test because of permissions, we would have noticed that.)

In particular, We already have [`trash`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/trash.rb), [`trash-cli`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/trash-cli.rb), and [`macos-trash`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/macos-trash.rb). All of them  serve a similar purpose, but do not require such permissions. (I installed them and tried trashing a file: `macos-trash` moved the file to `~/.local/share/Trash`, while the other two moved the file to the system-wide bin without asking me to grant permissions.)

> Would this then have to be a Cask instead?

It doesn't *have to*.

Given that its basic functionality can work without special permissions, I encourage you to consider asking upstream if this requirement can at least be loosened.

Open-source, CLI-only tools generally don't go to `homebrew/cask`, unless they are rejected by `homebrew/core` (see [the 4th bullet point](https://docs.brew.sh/Acceptable-Casks#rejected-casks)). But note that this formula doesn't have to end up rejected if its test is at least runnable on our runners.

**Comment by danielbayley** [comment]
> This pull request has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

Will fix…

**Comment by danielbayley** [comment]
> This pull request has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

Will fix…

**Comment by danielbayley** [comment]
> This pull request has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

Will fix…

**Comment by danielbayley** [comment]
> This pull request has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

Will fix…

**Comment by danielbayley** [comment]
> This pull request has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

Will fix…

**Pull request closed without merging**